syntax = "proto3";

import "graph.proto";

package router_serialize;

message RouteInternalData {
    double weight = 1;
    uint32 prev_edge_id = 2;
}

message VectorRouteInternalData {
    repeated RouteInternalData route_internal_data = 1;
}

message RoutesInternalData {
    repeated VectorRouteInternalData vector_route_internal_data = 1;
}

enum RouteActType {
    WAIT = 0;
    BUS = 1;
}

message RouteAct {
    RouteActType route_act_type = 1;
    string name = 2;
    double time = 3;
    uint32 span_count = 4;
    uint32 id = 5;
}

message TransportRouter {
    graph_serialize.Graph graph = 1;
    
    repeated RouteAct edge_id_to_route_act = 2;
}